### PR TITLE
#115755943  Create all the boilerplate Restful API endpoints for Suyabay.

### PR DESCRIPTION
**Pull request by olotintemitope**
#### What does this PR do?

Create all the boilerplate Restful API endpoints for Suyabay. 
#### Description of Task to be completed?
- Create user endpoints and the callback functions.
  - Get all users GET /api/v1/users
  - Get a single  user GET /api/v1/users/{id}
  - Post a user POST /api/v1/users
  - Update user PUT /api/v1/users/{id}
- Create channel endpoints and the callback functions
  - Get all channels GET /api/v1/channels
  - Get a single  channel GET /api/v1/channels/{id}
  - Post a channel POST /api/v1/channels
  - Update a channel PUT /api/v1/channels/{id}
- Create channel episodes endpoints and the callback functions.
  - Get all channel episodes GET /api/v1/channels/{channel_id}/episodes
  - Get a single channel episode  GET /api/v1/channels/{channel_id}/episodes/{episode_id}
  - Post a channel episode POST /api/v1/channels/{channel_id}/episodes
  - Update a channel PUT /api/v1/channels/{channel_id}/episodes/{episode_id}
- Create episode comments endpoints and the callback functions.
  - Get all episode comments GET /api/v1/episodes/{episode_id}/comments
  - Get a single  episode comment  GET /api/v1/episodes/{episode_id}/comments/{comment_id}
  - Post an episode comments  POST /api/v1/episodes/{episode_id}/comments
- Create user favourites endpoints and the callback functions
  - Get all user favourites GET /api/v1/users/{username}/favourites
  - Get a single favourite  GET /api/v1/users/{username}/favourites/{favourite_id}
  - Post a favourite  POST /api/v1/users/{username}/favourites
#### How should this be manually tested?
- You need to add Postman as an add-on extension to your browser
- Input the endpoint you want to test in the url section  e.g. /api/v1/users
- In the response body, you will be presented with a Json  e.g. {"message" : "return all users" } and the response header should contain the status code 200 OK.
#### What are the relevant pivotal tracker stories?
#115755943

### CODE REVIEW

**Comment by unicodeveloper** [comment]
Your answers for `Description of task to be Completed` are not correct. That question is referring to the current task you carried out on this PR. Please update your answers accordingly.

**Comment by olotintemitope** [comment]
@unicodeveloper I have effected the corrections on "Description of task to be Completed" for this PR.

File: tests/CommentTest.php [hunk 1]
```diff
@@ -102,6 +103,13 @@ public function testFetchComment()
         $this->assertArrayHasKey('comments', $oldCommCastedToArray);
 
     }
+=======
+    /**
+     * Assert that a HTTP GET request for fetch comment to the
+     *
+     * URL /fetchComment
+     */
+>>>>>>> Assert that fetch comments returns comments
```
Comment (unicodeveloper) [inline tests/CommentTest.php]: **Code Quality - Critical**

Why do we have a merge conflict here?

**Comment by unicodeveloper** [comment]
Please delete those commits that have those controller methods implemented from this PR. This PR should only contain commits for exactly what the story states which is the implementation of the routes @andela-tolotin

File: app/Http/routes.php [hunk 2]
```diff
@@ -108,15 +108,15 @@
...
-    'as'   => 'getuserfavourite'
-]);
+});
+
+Route::delete('api/v1/users/{username}/favourites', function() {
+    return json_encode(["message" => "allow a user to unfavourite an episodes"]);
+
+});
 
 /*
```
Comment (unicodeveloper) [inline app/Http/routes.php:134]: **Code Quality - Critical**

There is a lot of repetition. In every route, you are writing `api/v1/sdsd/dsds/`. You can take advantage of the route group and give all the routes a prefix of `api/v1` like so:

``` php
Route::group(['prefix' => 'api/v1/'], function() {
    Route::get('users', 'blahblabhblah');
    Route::get('users/{username}/favourites', 'blahblahblah');
});
```

**Comment by unicodeveloper** [comment]
What kind of description is this @andela-tolotin ?.. If you don't understand what to write there, then meet @andela-jkariuki to explain to you 

screen shot 2016-03-22 at 9 15 58 am

**Comment by olotintemitope** [comment]
@unicodeveloper I have effected the changes on description of task to be completed for  this PR #225.

**Comment by unicodeveloper** [comment]
This is also not correct. The question is talking about Manual testing not automated testing. So please give the appropriate answer to this question @andela-tolotin 

screen shot 2016-03-22 at 10 29 00 am

**Comment by unicodeveloper** [comment]
Why do you have some many NITPICK CI issues?..Please fix these issues and take note of why NitpickCi complains every time rather than wait for it to complain before you fix @andela-tolotin

**Comment by olotintemitope** [comment]
I have traced what Nitpick-CI complained about, I don't know where the message "Expected 1 space after FUNCTION keyword; 0 found" is coming from. I have also verify on style CI and every check passed.

**Comment by unicodeveloper** [comment]
It's simple. PSR-2 rule states that when you use a closure or callback, there should be a space after the `function` keyword. Check [here](http://www.php-fig.org/psr/psr-2/#6-closures). So it should be something like 

``` php

Route::get('episodes/{episode_id}/comments', function () {
...
```

NOT

``` PHP
Route::get('episodes/{episode_id}/comments', function()  {
...
```

@andela-tolotin

**Pull request merged**